tests/integration/pipeline/test_bootstrap_command.py

```diff
@@ -0,0 +1,227 @@
+            "1",  # Should we create ECR repo, 1 - No
+            "",  # Pipeline IP address range
+            "N",  # cancel
+        ]
+
+        bootstrap_process_execute = self.run_command_with_inputs(bootstrap_command_list, inputs)
+
+        self.assertEqual(bootstrap_process_execute.process.returncode, 0)
+        stdout = bootstrap_process_execute.stdout.decode()
+        self.assertTrue(stdout.strip().endswith("Should we proceed with the creation? [y/N]:"))
+
+    def test_interactive_with_some_required_resources_provided(self):
+        stage_name, stack_name = self._get_stage_and_stack_name()
+        self.stack_names = [stack_name]
+
+        bootstrap_command_list = self.get_bootstrap_command_list()
+
+        inputs = [
+            stage_name,
+            "arn:aws:iam::123:user/user-name",  # pipeline user
+            "arn:aws:iam::123:role/role-name",  # Pipeline execution role
+            "",  # CloudFormation execution role
+            "arn:aws:s3:::bucket-name",  # Artifacts bucket
+            "3",  # Should we create ECR repo, 3 - specify one
+            "arn:aws:ecr:::repository/repo-name",  # ecr repo
+            "1.2.3.4/24",  # Pipeline IP address range
+            "y",  # proceed
+        ]
+
+        bootstrap_process_execute = self.run_command_with_inputs(bootstrap_command_list, inputs)
+
+        self.assertEqual(bootstrap_process_execute.process.returncode, 0)
+        stdout = bootstrap_process_execute.stdout.decode()
+        self.assertIn("Successfully created!", stdout)
+        # make sure the not provided resource is the only resource created.
+        self.assertSetEqual({"CloudFormationExecutionRole"}, self._extract_created_resource_logical_ids(stack_name))
+
+    def test_interactive_pipeline_user_only_created_once(self):
+        """
+        Create 3 stages, only the first stage resource stack creates
+        a pipeline user, and the remaining two share the same pipeline user.
+        """
+        stage_names = []
+        for suffix in ["1", "2", "3"]:
+            stage_name, stack_name = self._get_stage_and_stack_name(suffix)
+            stage_names.append(stage_name)
+            self.stack_names.append(stack_name)
+
+        bootstrap_command_list = self.get_bootstrap_command_list()
+
+        for i, stage_name in enumerate(stage_names):
+            inputs = [
+                stage_name,
+                *([""] if i == 0 else []),  # pipeline user
+                "arn:aws:iam::123:role/role-name",  # Pipeline execution role
+                "arn:aws:iam::123:role/role-name",  # CloudFormation execution role
+                "arn:aws:s3:::bucket-name",  # Artifacts bucket
+                "1",  # Should we create ECR repo, 1 - No, 2 - Yes
+                "",  # Pipeline IP address range
+                "y",  # proceed
+            ]
+
+            bootstrap_process_execute = run_command_with_input(
+                bootstrap_command_list, ("\n".join(inputs) + "\n").encode()
```

Comment:
why are we joining in "\n" ? this is not the case with elsewhere

Reply:
This is how we used to do with `run_command_with_input`, see https://github.com/aws/aws-sam-cli/blob/5cc7680068c820e972d6165a0cccd21677e2a428/tests/integration/deploy/test_deploy_command.py#L535-L537

Reply:
I was trying to understand what is the difference between this one and every other place in this PR. I believe you just need to use `run_command_with_inputs` instead of `run_command_with_input`.

Also, consider moving `run_command_with_inputs` to testing_utils instead of defining it here only.

Reply:
good idea, moved to utils